src/parser/smt2/smt2.cpp

```diff
@@ -1019,15 +1019,14 @@ api::Term Smt2::applyParseOp(ParseOp& p, std::vector<api::Term>& args)
       isNeg = true;
       constVal = constVal[0];
     }
-    if (constVal.getKind() == api::DIVISION
-        && constVal[0].getKind() == api::CONST_RATIONAL
-        && constVal[1].getKind() == api::CONST_RATIONAL)
+    if (constVal.getKind() == api::DIVISION && isConstIntOrReal(constVal[0])
+        && isConstIntOrReal(constVal[1]))
     {
       std::stringstream sdiv;
       sdiv << (isNeg ? "-" : "") << constVal[0] << "/" << constVal[1];
       constVal = d_solver->mkReal(sdiv.str());
     }
-    else if (constVal.getKind() == api::CONST_RATIONAL && isNeg)
+    else if (isConstIntOrReal(constVal) && isNeg)
```

Comment:
Firstly: I think this only works if these values are integer.For rational values, the strings will not be properly formatted.
But secondly: why do we convert to string anyway???

Reply:
Converting to string is the only overflow safe method for constructing integers in the API.

Reply:
I've guarded for constant integers now.

There is a more general fix that I will apply in a later PR.